packages/typespec-client-generator-core/generated-defs/Azure.ClientGenerator.Core.ts

```diff
@@ -561,11 +563,21 @@ export type ClientNamespaceDecorator = (
  * ```typespec
  * op test(@param @alternateType(string) date: utcDateTime): void;
  * ```
+ * @example
+ * ```typespec
+ * model Test {
+ *   @alternateType(unknown)
+ *   thumbprint?: string;
+ *
+ *   @alternateType(AzureLocation[], "csharp")
+ *   locations: string[];
+ * }
+ * ```
  */
 export type AlternateTypeDecorator = (
   context: DecoratorContext,
   source: ModelProperty | Scalar,
-  alternate: Scalar,
+  alternate: Type,
```

Comment:
Is this generated? If so, why a `Scalar | Model | unknown` will generate a `Type`?

Reply:
It's generated part which we can't control, for the why part my bold guess is tcgc core think the latest base type for `Scalar | Model | unknown` is `Type`?

Reply:
I don't think we could put it aside. Could you try "if Scalar| Model what it procudes", "if unknown what it produces"?

Reply:
> I don't think we could put it aside. Could you try "if Scalar| Model what it procudes", "if unknown what it produces"?

`Scalar | Model` will produce `Scalar | Model` and single `unknown` will produce `Type`.